### DISCUSSION TITLE
Is possible to verify file without providing metal_id

### DISCUSSION THREAD

**Discussion by 4RSIM3R** [OPENSPHERE-Inc/metal-on-symbol#33] [Q&A] [answered]
Hello, I just wanna say thanks first to the developer who develops this library, because this SDK makes my project easier

But, it is possible to verify a file that previously `forge` without providing `metal_id` i just wanna some capabilities as `apostille` do.

Thank you :D

**Comment by hanatyan128** [comment, marked as answer]
I apologize for my long absence.

You can verify foreged metal without `metal_id` .
Instead, use `source account`, `target account`, `target ID` (mosaic / namespace only) and file.
Because the `metal_id` is calculated from the infomation above.

**Example code here**

```
// Calculate metadata key (If metadata key is known, you can omit this)
const metadataKey = await MetalService.calculateMetadataKey(fileData);

// Verify on-chain data (`targetId` is `undefined` when the metal was forged to account)
const { mismatches } = await metalService.verify(fileData, metadataType, sourceAddress, targetAddress, metadataKey , targetId)

assert(mismatches === 0);
```
